engine/src/main/java/org/pentaho/di/kitchen/Kitchen.java

```diff
@@ -0,0 +1,372 @@
+              .logLevel( "" )
+              .maxLogLines( "" )
+              .maxLogTimeout( "" )
+              .logFile( "" )
+              .oldLogFile( "" )
+              .version( "" )
+              .resultSetStepName( "" )
+              .resultSetCopyNumber( "" )
+              .base64Zip( optionBase64Zip.toString() )
+              .namedParams( optionParams )
+              .customNamedParams( customOptions )
+              .build();
+
+      result = getCommandExecutor().execute( jobParams, args.toArray( new String[ args.size() ] ) );
+
+    } catch ( Throwable t ) {
+      t.printStackTrace();
+      result.setExitStatus( CommandExecutorCodes.Pan.UNEXPECTED_ERROR.getCode() );
+
+    } finally {
+      if ( fileAppender != null ) {
+        fileAppender.close();
+        KettleLogStore.getAppender().removeLoggingEventListener( fileAppender );
+      }
+    }
+
+    exitJVM( result.getExitStatus() );
+
+  }
+
+  private static <T extends Throwable> void blockAndThrow( Future<T> future ) throws T {
+    try {
+      T e = future.get();
+      if ( e != null ) {
+        throw e;
+      }
+    } catch ( InterruptedException e ) {
+      throw new RuntimeException( e );
+    } catch ( ExecutionException e ) {
+      throw new RuntimeException( e );
+    }
+  }
+
+  /**
+   * Configure the central log store from the provided command line options
+   *
+   * @param maxLogLinesOption   Option for maximum log lines
+   * @param maxLogTimeoutOption Option for log timeout
+   * @throws KettleException Error parsing command line arguments
+   */
+  public static void configureLogging( final CommandLineOption maxLogLinesOption,
+                                       final CommandLineOption maxLogTimeoutOption ) throws KettleException {
+    int maxLogLines = parseIntArgument( maxLogLinesOption, 0 );
+    if ( Utils.isEmpty( maxLogLinesOption.getArgument() ) ) {
+      maxLogLines = Const.toInt( EnvUtil.getSystemProperty( Const.KETTLE_MAX_LOG_SIZE_IN_LINES ), 5000 );
+    }
+    int maxLogTimeout = parseIntArgument( maxLogTimeoutOption, 0 );
+    if ( Utils.isEmpty( maxLogTimeoutOption.getArgument() ) ) {
+      maxLogTimeout = Const.toInt( EnvUtil.getSystemProperty( Const.KETTLE_MAX_LOG_TIMEOUT_IN_MINUTES ), 1440 );
+    }
+    KettleLogStore.init( maxLogLines, maxLogTimeout );
+  }
+
+  /**
+   * Parse an argument as an integer.
+   *
+   * @param option Command Line Option to parse argument of
+   * @param def    Default if the argument is not set
+   * @return The parsed argument or the default if the argument was not specified
+   * @throws KettleException Error parsing provided argument as an integer
+   */
+  protected static int parseIntArgument( final CommandLineOption option, final int def ) throws KettleException {
+    if ( !Utils.isEmpty( option.getArgument() ) ) {
+      try {
+        return Integer.parseInt( option.getArgument().toString() );
+      } catch ( NumberFormatException ex ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)